### ISSUE TITLE
Majority if demos broken on Oculus Quest

### ISSUE THREAD

**Issue by vedtam** [aframevr/aframe#4332]
Hi,

While trying to test some of the demos found on the a-frame homepage, i wasn't able to run any of them. I used Mozilla Reality to run the demos on my Quest. For example Moon Rider starts but as I try to hit/smash any box that comes towards you the game instantly freezes.

I'm kinda new to webvr, wondering if these VR headset are capable of running Web VR?

**Comment by dmarcos** [comment]
@vedtam All demos should run fine. Did you have any issues with the Oculus Browser?

**Comment by vedtam** [comment]
Hi,

Tried in the oculus browser, first it started in 2D than in the bottom right I selected VR 360* which distorted the view by blowing things up and started lagging seriously. Hands and objects weren't rendering at all. :(

Any other quest users having issues?

**Comment by dmarcos** [comment]
FWIW, all official [A-Frame examples](https://aframe.io/aframe/examples/) work for me on the Oculus browser. Also [Mozilla Hubs](https://hubs.mozilla.com/) and [Moonrider](http://moonrider.xyz/) and [Supercraft](https://supermedium.com/craft/) work fine. Can you share the specific links that are not working?

**Comment by vedtam** [comment]
Hi,

Created a few screenshots and a video recording (maybe I'm doing something wrong:

![com oculus vrshell-20191127-103738](https://user-images.githubusercontent.com/4981592/69707792-0f732580-1103-11ea-8cd0-62e7ed0f51a9.jpg)
![com oculus vrshell-20191127-103858](https://user-images.githubusercontent.com/4981592/69707796-126e1600-1103-11ea-98c1-58065d146650.jpg)
![com oculus vrshell-20191127-103516](https://user-images.githubusercontent.com/4981592/69707824-1bf77e00-1103-11ea-8e04-148089764f02.jpg)
![com oculus vrshell-20191127-103557](https://user-images.githubusercontent.com/4981592/69707829-1e59d800-1103-11ea-80c8-fb8d58654b8f.jpg)
![com oculus vrshell-20191127-103613](https://user-images.githubusercontent.com/4981592/69707831-20239b80-1103-11ea-833d-708a60355a0e.jpg)

and the video:
https://a.uguu.se/wqEGPTDnhgFV_com.oculus.vrshell-20191127-103753.mp4

**Comment by dmarcos** [comment]
Are you using the Oculus Browser? Below a gif of [moonrider](http://moonrider.xyz/) entering VR as expected (notice you have to click on the Goggles icon in the bottom right corner)

![caca](https://user-images.githubusercontent.com/39342/69738691-7b09c280-1104-11ea-83da-d0520886d2fa.gif)

**Comment by dmarcos** [comment]
Below Firefox Reality also entering VR as expected:
![caca](https://user-images.githubusercontent.com/39342/69739234-60841900-1105-11ea-8be9-3cd2a3643548.gif)

**Comment by dmarcos** [comment]
Moonrider gameplay video on Oculus Browser:

https://drive.google.com/file/d/15CUgmosIe4eu64nrWWinPr3_wqHk3i9e/view

**Comment by vedtam** [comment]
Hi,

Thanks, surely I access the vr more via the glasses icon in the bottom right corner. And yes, I've tested bot on the default browser and Firefox reality. Same happens. Could you share your Quest OS version?

**Comment by dmarcos** [comment]
OS Version: 415630.6700.0

We have a big community playing moonrider and haven't heard of any issues recently.

**Comment by vedtam** [comment]
Thanks, i have the same, but something must be different. I've tried switching Firefox to desktop mode, now the scene looks fine, but as I hit the first object flying towards me, the app freezes but the music continues to play. Did a screen recording, will eventually upload it later or should I rather post it on Moonrider's repo?

**Comment by dmarcos** [comment]
If it’s a problem that only affects Firefox Reality probably better to open an issue in [their repo](https://github.com/MozillaReality/FirefoxReality/issues)

**Comment by vedtam** [comment]
It affects both Firefox and the Oculus (chromium) browsers. Will try to debug first and gather some analysis if possible.

**Comment by vedtam** [comment]
Hi,

I figured it out! Few notes (minor): 

When opening the A-frame homepage, with default browser settings, Moon Rider and a few other examples are missing from the left side (checked the markup, haven't seen any media query that would hide it for whatever reason, but probably its on purpose), however as I set the Oculus Browser to desktop mode, Moon Rider (and others) shows up - 0:10s in video.

Now if I open Moon Rider (or other examples) and click the glasses icon to enter VR mode, I get only a 2D scree, if switching to VR 360 (bottom right) the scene gets distorted and the game freezes upon playing.

"Solution":

However if I navigate back and open Showcase section and run Moon Rider there, the game plays flawlessly! (man its such a nice game / execution!).

Video:

https://drive.google.com/file/d/1St8TbIM1cgBTr7ACFvCZ8IihTqfh28lf/view
